### PR TITLE
[Core v1.12 beta2]Adding parser docs

### PR DESCRIPTION
**Pull request by runleonarun**
## What are you changing in this pull request and why?

* Addresses [dev PR](https://github.com/dbt-labs/dbt-core/pull/13029)
* Adds to upgrade guide and  

## Checklist

- [ ] The changes in this PR meet the [docs style guide/fundamentals](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) required for all content.
- [ ] Applied the proper versioning rules if the content is for specific dbt version(s): ([version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) or [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content)
- [ ] The content in this PR requires a dbt release note, so I added one to the [release notes page](https://docs.getdbt.com/docs/dbt-versions/dbt-cloud-release-notes).).

---
🚀 Deployment available! Here are the direct links to the updated files:

- https://docs-getdbt-com-git-leona-use-v2-parser-docs-dbt-labs.vercel.app/docs/dbt-versions/core-upgrade/02-upgrading-to-fusion
- https://docs-getdbt-com-git-leona-use-v2-parser-docs-dbt-labs.vercel.app/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.12
- https://docs-getdbt-com-git-leona-use-v2-parser-docs-dbt-labs.vercel.app/reference/global-configs/about-global-configs
- https://docs-getdbt-com-git-leona-use-v2-parser-docs-dbt-labs.vercel.app/reference/global-configs/parsing

### CODE REVIEW

File: website/docs/reference/global-configs/parsing.md [hunk 1]
```diff
@@ -54,6 +54,60 @@ config:
...
+
+<File name="Usage">
+
+```bash
+dbt run --use-v2-parser --v2-parser "uv run fs parse"
+```
+
+</File>
+
+When the v2 parser fails, dbt surfaces typed exceptions to make failures easier to diagnose:
```
Comment (runleonarun) [inline website/docs/reference/global-configs/parsing.md]:
```suggestion
When the v2 parser fails, dbt surfaces these exceptions to make failures easier to diagnose:
```

File: website/docs/reference/global-configs/about-global-configs.md [hunk 2]
```diff
@@ -156,6 +156,8 @@ Use this table to compare all available flags and how to configure them across i
 | [use_colors_file](/reference/global-configs/logs#color) | ❌ | boolean <br /> default: True | ✅ | `DBT_ENGINE_USE_COLORS_FILE` | `--use-colors-file` <br /> `--no-use-colors-file` |
 | [use_colors](/reference/global-configs/print-output#print-color) | ❌ | boolean <br /> default: True | ✅ | `DBT_ENGINE_USE_COLORS` | `--use-colors` <br /> `--no-use-colors` |
 | [use_experimental_parser](/reference/global-configs/parsing#experimental-parser) | ❌ | boolean <br /> default: False | ✅ | `DBT_ENGINE_USE_EXPERIMENTAL_PARSER` | `--use-experimental-parser` <br /> `--no-use-experimental-parser` |
+| [use_v2_parser](/reference/global-configs/parsing#opt-in-v2-parser) | ✅ | boolean <br /> default: False | ✅ | `DBT_ENGINE_USE_V2_PARSER` | `--use-v2-parser` |
+| [v2_parser](/reference/global-configs/parsing#opt-in-v2-parser) | ✅ | string <br /> default: `fs parse` | ❌ | `DBT_ENGINE_V2_PARSER` | `--v2-parser` |
```
Comment (aiguofer) [inline website/docs/reference/global-configs/about-global-configs.md]: This flag isn't technically public and users likely would never need to use it. We mainly have it for testing with latest fusion builds, do we normally publish non-public flags?

File: website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.12.md [hunk 3]
```diff
@@ -56,6 +56,12 @@ When you use the legacy `--selector` flag together with `--select` or `--exclude
 
 You can use the [`vars.yml`](/docs/build/project-variables#defining-variables-in-varsyml) file, located at the project root, to define project variables. This keeps variable definitions in one place and helps simplify `dbt_project.yml`. Variables defined in `vars.yml` are parsed _before_ `dbt_project.yml`, so you can reference them in `dbt_project.yml` using `{{ var('...') }}`. You can continue to define variables in `dbt_project.yml` as before, but you cannot define variables in both files. For details and precedence, refer to [Project variables](/docs/build/project-variables).
 
+### Opt-in v2 parser <Lifecycle status="beta" />
+
+<Constant name="core" /> v1.12 introduces the `--use-v2-parser` flag, which delegates parsing to the Fusion parser (`fs parse`) instead of dbt Core's own parser. This is an opt-in flag that changes no behavior unless explicitly set, making it a low-risk way to test Fusion parser compatibility from within <Constant name="core" /> v1.12.
```
Comment (aiguofer) [inline website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.12.md]: binary is no longer `fs`... and binary name shouldn't really matter to users. maybe we can just omit that?

File: website/docs/reference/global-configs/parsing.md [hunk 4]
```diff
@@ -54,6 +54,60 @@ config:
...
+```yaml
+flags:
+  use_v2_parser: true
+```
+
+</File>
+
+Other behaviors to know about include:
+
+- **`v2_parser` (custom parse command)**: By default, dbt invokes `fs parse` when `--use-v2-parser` is set. To use a custom command or wrapper, set `--v2-parser` (env: `DBT_ENGINE_V2_PARSER`).
```
Comment (aiguofer) [inline website/docs/reference/global-configs/parsing.md]: Hmm, I wonder if we actually want to cover this. This is technically a hidden flag used mainly by us for testing. I can't see much value for users and could be confusing, but it could let them use the full fusion binary if they have it...

**Comment by aiguofer** [comment]
I feel like we should address potential issues:

The manifest that Fusion generates can have some differences to the manifest that Core generates. This could alter behavior in Core under certain circumstances. We're working on closing the gap on manifest differences, but this is beta and users should be aware that there are likely unknown bugs at this point.

File: website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.12.md (hunk 3 above)
Comment (runleonarun) [inline website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.12.md]:
```suggestion
<Constant name="core" /> v1.12 introduces the `--use-v2-parser` flag that delegates parsing to the Fusion parser instead of dbt Core's own parser. This is an opt-in flag that changes no behavior unless explicitly set, making it a low-risk way to test Fusion parser compatibility from within <Constant name="core" /> v1.12.
```

File: website/docs/reference/global-configs/about-global-configs.md (hunk 2 above)
Comment (runleonarun) [inline website/docs/reference/global-configs/about-global-configs.md]:
```suggestion
```

Comment (runleonarun) [reply website/docs/reference/global-configs/about-global-configs.md]: Ah good. Maybe that's why it was confusing to me.

File: website/docs/reference/global-configs/parsing.md [hunk 5]
```diff
@@ -54,6 +54,60 @@ config:
 
 </File>
 
+### Opt-in v2 parser {#opt-in-v2-parser}
+
+<VersionBlock firstVersion="1.12" lastVersion="1.99">
+
+The `use_v2_parser` flag delegates parsing to the Fusion parser (`fs parse`) instead of <Constant name="core" />'s own parser. This is an opt-in flag — it changes no behavior unless explicitly set.
```
Comment (runleonarun) [inline website/docs/reference/global-configs/parsing.md]:
```suggestion
The `use_v2_parser` flag delegates parsing to the Fusion parser instead of <Constant name="core" />'s own parser. This is an opt-in flag — it changes no behavior unless explicitly set.
```

File: website/docs/reference/global-configs/parsing.md (hunk 4 above)
Comment (runleonarun) [inline website/docs/reference/global-configs/parsing.md]:
```suggestion
```

File: website/docs/reference/global-configs/parsing.md [hunk 6]
```diff
@@ -54,6 +54,60 @@ config:
...
+- **`v2_parser` (custom parse command)**: By default, dbt invokes `fs parse` when `--use-v2-parser` is set. To use a custom command or wrapper, set `--v2-parser` (env: `DBT_ENGINE_V2_PARSER`).
+
+- **Partial parsing**: Partial parsing is disabled when `--use-v2-parser` is set. Any stale `partial_parse.msgpack` from a prior run is automatically removed on entry.
+- **`write_manifest`**: `write_manifest` does not work in this mode because the Fusion parser's artifacts (`manifest.json` and `semantic_manifest.json`) are canonical and dbt Core does not re-serialize or overwrite them.
+- **Artifacts in `target/`**: When `write_json` is enabled, the handoff `manifest.json` (and `semantic_manifest.json` if present) is copied into your project's `target/` directory.
+
+<File name="Usage">
+
+```bash
+dbt run --use-v2-parser --v2-parser "uv run fs parse"
```
Comment (runleonarun) [inline website/docs/reference/global-configs/parsing.md]: @aiguofer what should `"uv run fs parse"` say? or can we drop it? Is it just a comment in the logs?

File: website/docs/reference/global-configs/parsing.md [hunk 7]
```diff
@@ -54,6 +54,59 @@ config:
...
+```
+
+</File>
+
+When the v2 parser fails, dbt surfaces these exceptions to make failures easier to diagnose:
+
+| Exception | Cause |
+|---|---|
+| `FusionParserMissingError` | The parser binary was not found |
```
Comment (runleonarun) [inline website/docs/reference/global-configs/parsing.md]:
```suggestion
| `FusionParserMissingError` | The parser was not found |
```

File: website/docs/reference/global-configs/parsing.md [hunk 8]
```diff
@@ -54,6 +54,59 @@ config:
...
+dbt run --use-v2-parser --v2-parser "uv run fs parse"
+```
+
+</File>
+
+When the v2 parser fails, dbt surfaces these exceptions to make failures easier to diagnose:
+
+| Exception | Cause |
+|---|---|
+| `FusionParserMissingError` | The parser was not found |
```
Comment (runleonarun) [inline website/docs/reference/global-configs/parsing.md]: hey @aiguofer What should users do if the parser was not found?

File: website/docs/reference/global-configs/parsing.md [hunk 9]
```diff
@@ -54,6 +54,59 @@ config:
...
+| `FusionParserError` | The parser exited with a non-zero code or produced no output |
+| `FusionParserSchemaError` | The output could not be parsed as valid JSON |
```
Comment (runleonarun) [inline website/docs/reference/global-configs/parsing.md]:
```suggestion
| `FusionParserSchemaError` | The output could not be parsed as valid JSON. Please check your JSON and try again. |
```

File: website/docs/reference/global-configs/parsing.md [hunk 10]
```diff
@@ -54,6 +54,59 @@ config:
...
+| `FusionParserVersionError` | The output manifest uses an incompatible schema version |
```
Comment (runleonarun) [inline website/docs/reference/global-configs/parsing.md]:
```suggestion
| `FusionParserVersionError` | The output manifest uses an incompatible schema version. Make sure you're using Core v1.12 or higher. |
```

File: website/docs/reference/global-configs/parsing.md [hunk 11]
```diff
@@ -54,6 +54,59 @@ config:
...
+```
+
+</File>
+
+When the v2 parser fails, dbt surfaces these exceptions to make failures easier to diagnose:
+
+| Exception | Cause |
+|---|---|
+| `FusionParserMissingError` | The parser was not found |
+| `FusionParserError` | The parser exited with a non-zero code or produced no output |
```
Comment (runleonarun) [inline website/docs/reference/global-configs/parsing.md:97]: @aiguofer what should users do for this one too?

File: website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.12.md [hunk 12]
```diff
@@ -56,6 +56,12 @@ When you use the legacy `--selector` flag together with `--select` or `--exclude
 
 You can use the [`vars.yml`](/docs/build/project-variables#defining-variables-in-varsyml) file, located at the project root, to define project variables. This keeps variable definitions in one place and helps simplify `dbt_project.yml`. Variables defined in `vars.yml` are parsed _before_ `dbt_project.yml`, so you can reference them in `dbt_project.yml` using `{{ var('...') }}`. You can continue to define variables in `dbt_project.yml` as before, but you cannot define variables in both files. For details and precedence, refer to [Project variables](/docs/build/project-variables).
 
+### Opt-in v2 parser <Lifecycle status="beta" />
+
+<Constant name="core" /> v1.12 introduces the `--use-v2-parser` flag that delegates parsing to the Fusion parser instead of dbt Core's own parser. This is an opt-in flag that changes no behavior unless explicitly set, making it a low-risk way to test Fusion parser compatibility from within <Constant name="core" /> v1.12.
+
```
Comment (runleonarun) [inline website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.12.md:76]:
```suggestion
:::note
The Rust parser is beta. Its output manifest may differ from the Python parser's in edge cases, which can affect downstream behavior. Fall back by removing `--use-v2-parser`, and [report issues]() to help us close the gap.
:::
```

File: website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.12.md [hunk 13]
```diff
@@ -56,6 +56,12 @@ When you use the legacy `--selector` flag together with `--select` or `--exclude
 
 You can use the [`vars.yml`](/docs/build/project-variables#defining-variables-in-varsyml) file, located at the project root, to define project variables. This keeps variable definitions in one place and helps simplify `dbt_project.yml`. Variables defined in `vars.yml` are parsed _before_ `dbt_project.yml`, so you can reference them in `dbt_project.yml` using `{{ var('...') }}`. You can continue to define variables in `dbt_project.yml` as before, but you cannot define variables in both files. For details and precedence, refer to [Project variables](/docs/build/project-variables).
 
+### Opt-in v2 parser <Lifecycle status="beta" />
+
+<Constant name="core" /> v1.12 introduces the `--use-v2-parser` flag that delegates parsing to the Fusion parser instead of dbt Core's own parser. This is an opt-in flag that changes no behavior unless explicitly set, making it a low-risk way to test Fusion parser compatibility from within <Constant name="core" /> v1.12.
```
Comment (runleonarun) [inline website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.12.md]:
```suggestion
<Constant name="core" /> v1.12 introduces the `--use-v2-parser` flag that delegates parsing to Fusion's Rust parser instead of dbt Core's own Python parser. This is an opt-in flag that changes no behavior unless explicitly set, making it a low-risk way to test Fusion parser compatibility from within <Constant name="core" /> v1.12.
```

File: website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.12.md [hunk 14]
```diff
@@ -56,6 +56,14 @@ When you use the legacy `--selector` flag together with `--select` or `--exclude
 
 You can use the [`vars.yml`](/docs/build/project-variables#defining-variables-in-varsyml) file, located at the project root, to define project variables. This keeps variable definitions in one place and helps simplify `dbt_project.yml`. Variables defined in `vars.yml` are parsed _before_ `dbt_project.yml`, so you can reference them in `dbt_project.yml` using `{{ var('...') }}`. You can continue to define variables in `dbt_project.yml` as before, but you cannot define variables in both files. For details and precedence, refer to [Project variables](/docs/build/project-variables).
 
+### Opt-in v2 parser <Lifecycle status="beta" />
+
+<Constant name="core" /> v1.12 introduces the `--use-v2-parser` flag that delegates parsing to Fusion's Rust parser instead of dbt Core's own Python parser. This is an opt-in flag that changes no behavior unless explicitly set, making it a low-risk way to test Fusion parser compatibility from within <Constant name="core" /> v1.12.
+:::note
```
Comment (runleonarun) [inline website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.12.md:78]:
```suggestion

:::note
```

File: website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.12.md [hunk 15]
```diff
@@ -56,6 +56,16 @@ When you use the legacy `--selector` flag together with `--select` or `--exclude
...
 
+### Opt-in v2 parser <Lifecycle status="beta" />
+
+<Constant name="core" /> v1.12 introduces the `--use-v2-parser` flag that delegates parsing to Fusion's Rust parser instead of dbt Core's own Python parser. This is an opt-in flag that changes no behavior unless explicitly set, making it a low-risk way to test Fusion parser compatibility from within <Constant name="core" /> v1.12.
+
+
+:::note
+The Rust parser is beta. Its output manifest may differ from the Python parser's in edge cases, which can affect downstream behavior. Fall back by removing `--use-v2-parser`, and [report issues]() to help us close the gap.
+:::
```
Comment (runleonarun) [inline website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.12.md:80]:
```suggestion
:::

```

File: website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.12.md [hunk 16]
```diff
@@ -56,6 +56,16 @@ When you use the legacy `--selector` flag together with `--select` or `--exclude
...
+For how to enable the flag, related behaviors, and parser error types, refer to [Opt-in v2 parser](/reference/global-configs/parsing#opt-in-v2-parser).
```
Comment (runleonarun) [inline website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.12.md]:
```suggestion
For more information on how to enable the flag, related behaviors, and parser error types, refer to [Opt-in v2 parser](/reference/global-configs/parsing#opt-in-v2-parser).
```

File: website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.12.md [hunk 17]
```diff
@@ -56,6 +56,17 @@ When you use the legacy `--selector` flag together with `--select` or `--exclude
 
 You can use the [`vars.yml`](/docs/build/project-variables#defining-variables-in-varsyml) file, located at the project root, to define project variables. This keeps variable definitions in one place and helps simplify `dbt_project.yml`. Variables defined in `vars.yml` are parsed _before_ `dbt_project.yml`, so you can reference them in `dbt_project.yml` using `{{ var('...') }}`. You can continue to define variables in `dbt_project.yml` as before, but you cannot define variables in both files. For details and precedence, refer to [Project variables](/docs/build/project-variables).
 
+### Opt-in v2 parser <Lifecycle status="beta" />
+
+<Constant name="core" /> v1.12 introduces the `--use-v2-parser` flag that delegates parsing to Fusion's Rust parser instead of dbt Core's own Python parser. This is an opt-in flag that changes no behavior unless explicitly set, making it a low-risk way to test Fusion parser compatibility from within <Constant name="core" /> v1.12.
```
Comment (runleonarun) [inline website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.12.md]:
```suggestion
<Constant name="core" /> v1.12 introduces the `--use-v2-parser` flag that delegates parsing to Fusion's Rust parser instead of dbt Core's own Python parser. The Rust parser is significantly faster than the Python parser — especially on larger projects, where it can be 5–10× quicker. If you're looking to speed up your development workflow or cut down on job startup times. Using the Rust parser is a natural first step toward Fusion compatibility, so you can catch and fix any project issues gradually rather than all at once.

This is an opt-in flag that changes no behavior unless explicitly set, making it a low-risk way to test Fusion parser compatibility from within <Constant name="core" /> v1.12.
```

File: website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.12.md [hunk 18]
```diff
@@ -56,6 +56,19 @@ When you use the legacy `--selector` flag together with `--select` or `--exclude
 
 You can use the [`vars.yml`](/docs/build/project-variables#defining-variables-in-varsyml) file, located at the project root, to define project variables. This keeps variable definitions in one place and helps simplify `dbt_project.yml`. Variables defined in `vars.yml` are parsed _before_ `dbt_project.yml`, so you can reference them in `dbt_project.yml` using `{{ var('...') }}`. You can continue to define variables in `dbt_project.yml` as before, but you cannot define variables in both files. For details and precedence, refer to [Project variables](/docs/build/project-variables).
 
+### Opt-in v2 parser <Lifecycle status="beta" />
+
+<Constant name="core" /> v1.12 introduces the `--use-v2-parser` flag that delegates parsing to Fusion's Rust parser instead of dbt Core's own Python parser. The Rust parser is significantly faster than the Python parser — especially on larger projects, where it can be 5–10× quicker. If you're looking to speed up your development workflow or cut down on job startup times. Using the Rust parser is a natural first step toward Fusion compatibility, so you can catch and fix any project issues gradually rather than all at once.
```
Comment (runleonarun) [inline website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.12.md:73]: hey @aiguofer I also added this info on the parser. LMK if it needs help.

File: website/docs/reference/global-configs/parsing.md (hunk 6 above)
Comment (aiguofer) [reply website/docs/reference/global-configs/parsing.md]: I think we can just drop it

File: website/docs/reference/global-configs/parsing.md (hunk 8 above)
Comment (aiguofer) [reply website/docs/reference/global-configs/parsing.md]: Ohh I didn't catch this earlier, we actually removed this error!

File: website/docs/reference/global-configs/parsing.md (hunk 11 above)
Comment (aiguofer) [reply website/docs/reference/global-configs/parsing.md:97]: hmm this is the most generic.... it's hard to tell because it could just be an error to parse in Fusion, in which case fusion itself will provide feedback on what needs fixing... but it could also be something gone wrong with the binary or setup in general, in which case it could be more involved.

File: website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.12.md (hunk 18 above)
Comment (aiguofer) [reply website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.12.md:73]: that's great!

File: website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.12.md [hunk 19]
```diff
@@ -56,6 +56,19 @@ When you use the legacy `--selector` flag together with `--select` or `--exclude
...
+
+This is an opt-in flag that changes no behavior unless explicitly set, making it a low-risk way to test Fusion parser compatibility from within <Constant name="core" /> v1.12.
+
+
+:::note
+The Rust parser is beta. Its output manifest may differ from the Python parser's in edge cases, which can affect downstream behavior. Fall back by removing `--use-v2-parser`, and [report issues]() to help us close the gap.
```
Comment (runleonarun) [inline website/docs/docs/dbt-versions/core-upgrade/03-upgrading-to-v1.12.md]:
```suggestion
The Rust parser is beta. Its output manifest may differ from the Python parser's in edge cases, which can affect downstream behavior. Fall back by removing `--use-v2-parser`, and [report issues](https://github.com/dbt-labs/dbt-core/issues) to help us close the gap.
```

File: website/docs/reference/global-configs/parsing.md (hunk 8 above)
Comment (runleonarun) [inline website/docs/reference/global-configs/parsing.md]:
```suggestion
```

File: website/docs/reference/global-configs/parsing.md [hunk 20]
```diff
@@ -54,6 +54,58 @@ config:
...
+
+
+- **Partial parsing**: Partial parsing is disabled when `--use-v2-parser` is set. Any stale `partial_parse.msgpack` from a prior run is automatically removed on entry.
+- **`write_manifest`**: `write_manifest` does not work in this mode because the Fusion parser's artifacts (`manifest.json` and `semantic_manifest.json`) are canonical and dbt Core does not re-serialize or overwrite them.
+- **Artifacts in `target/`**: When `write_json` is enabled, the handoff `manifest.json` (and `semantic_manifest.json` if present) is copied into your project's `target/` directory.
+
+<File name="Usage">
+
+```bash
+dbt run --use-v2-parser --v2-parser "uv run fs parse"
```
Comment (runleonarun) [inline website/docs/reference/global-configs/parsing.md]:
```suggestion
dbt run --use-v2-parser
```

File: website/docs/reference/global-configs/parsing.md (hunk 11 above)
Comment (runleonarun) [reply website/docs/reference/global-configs/parsing.md:97]: @aiguofer should I tell them to open an issue in dbt-core if they can't solve the issue?

Comment (mirnawong1) [reply website/docs/reference/global-configs/parsing.md:97]: we can address as fast follow

**Review by mirnawong1 — APPROVED** [review]
